trl/trainer/callbacks.py

```diff
@@ -233,3 +235,86 @@ def on_evaluate(self, args: TrainingArguments, state: TrainerState, control: Tra
         if self.trainer.accelerator.is_main_process:
             win_rate = sum(winner_idx == 1 for winner_idx in winner_indices) / len(winner_indices)
             self.trainer.log({"eval_win_rate": win_rate})
+
+
+class LogCompletionsCallback(WandbCallback):
+    r"""
+    A [`~transformers.TrainerCallback`] that logs completions to Weights & Biases.
+
+    Usage:
+    ```python
+    prompts = ["The capital of France is", "The opposite of up is"]
+    trainer = DPOTrainer(..., callbacks=[LogCompletionsCallback(prompts)])
+    ```
+
+    Args:
+        prompts (`List[str]`):
+            The prompts to generate completions for.
+        freq (`Optional[int]`, *optional*, defaults to `None`):
+            The frequency at which to log completions. If not provided, defaults to `logging_steps`.
```

Comment:
This was previously set as `save_steps` for logging, but I think `logging_steps` is more intuitive

Reply:
Yes, but I think you have to be careful here:
- Unlike datapoints, having more than 10 completion batches is completely indigestible, and the number of datapoints for a run is always well over 10.
- WandB tables have a limit of 10,000 rows. For 8 prompts, this means that the run will fail after ~1100 logging steps. 
- Also, I haven't found a way of adding a row, so I have to upload the complete updated table each time. The way wandb works internally is that it stores all logged tables as artifacts. Hence, each run therefore could become very large.

All this to say that I don't know what the best option is here

Reply:
I see, I guess for long runs the user can tweak `logging_steps` accordingly. Let's leave it like this for now and work on a fix if issues are opened